core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -468,6 +494,9 @@ public void closed(final Status status, final Metadata.Trailers trailers) {
           @Override
           public void run() {
             closed = true;
+            if (deadlineCancellationFuture != null) {
```

Comment:
`volatile` reads aren't optimized, so ideally you would read `deadlineCancellationFuture` into a local variable. It's fine as-is, but it is also good practice to use the local variable.